deepface/modules/preprocessing.py

```diff
@@ -129,3 +143,58 @@ def normalize_input(img: np.ndarray, normalization: str = "base") -> np.ndarray:
         raise ValueError(f"unimplemented normalization type - {normalization}")
 
     return img
+
+
+def extract_frames_with_faces(video_bytes):
+  """Extracts frames with faces from a video as base64 bytes.
+
+  Args:
+    video_bytes: The video bytes.
+
+  Returns:
+    A list of base64-encoded frames with faces.
+  """
+  video_bytes = base64.b64decode(video_bytes)
+
+  # Create a file-like object to write the video to.
+  hash = str(random.getrandbits(128))
+  with io.open("./"+ hash + ".mp4", "wb") as f:
+    f.write(video_bytes)
+  # Decode the video bytes.
+  video_capture = cv2.VideoCapture("./"+ hash + ".mp4")
+
+  # Create a face detector.
+  face_detector = cv2.CascadeClassifier(cv2.data.haarcascades + 'haarcascade_frontalface_default.xml')
+
+  # Extract frames with faces.
+  frames_with_faces = []
+  tempFrame = None
+  while True:
+    ret, frame = video_capture.read()
+    if not ret:
+        frame_bytes = cv2.imencode('.jpg', tempFrame)[1].tobytes()
+        frame_base64 = base64.b64encode(frame_bytes).decode('utf-8')
+        frames_with_faces.append(frame_base64)
+        break
+    else:
+        tempFrame = frame
+
+    # Convert the frame to grayscale.
+    gray = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
+
+    # Detect faces in the frame.
+    faces = face_detector.detectMultiScale(gray, 1.3, 5)
```

Comment:
once load image is called, face detector module is being called already. why we need to call face extraction here?

https://github.com/serengil/deepface/blob/master/deepface/modules/detection.py#L84